cleanup: use better-maintained sha-1 crate instead of sha1 crate

durin42 opened the pull request:
This removes a "duplicate" dependency in the tree of pyoxidizer.

alex commented:
sha1 has a higher MSRV than UUID currently does.

newpavlov commented:
`sha-1 v0.8` has MSRV [equal to 1.21](https://github.com/RustCrypto/hashes/tree/sha1-v0.8.2#minimum-supported-rust-version-msrv). Ideally together with such change it's worth to replace `md5` with `md-5` as well, though maintainers have previously [rejected](https://github.com/uuid-rs/uuid/pull/93) my PR with such changes. :/

KodrAus commented:
It looks like the original concern came from the weight of those dependencies? I had a look through them and they didn't seem too weighty to me.

I'd be in favour of also bundling `md-5` into this PR if you're happy to do that.

durin42 commented:
It was mostly I noticed the two implementations of SHA1 as I was importing crates into a monorepo and figured I should try and consolidate down onto the maintained-looking one. If you're raising your MSRV I'm happy to handle both sha1 and md5, but if you want to keep it low please feel encouraged to reject this PR.

Thanks!

KodrAus commented:
We've actually already raised it, so please feel free to roll in both!

KodrAus commented:
@durin42 I just pushed a quick merge commit to your branch. Hope you don't mind!

durin42 commented:
Not at all! I'll try and get the md5 upgrade done too over the weekend. Right now my keyboard time is somewhat limited. :)

durin42 commented:
I've rebased and added a commit for upgrading md5 to md-5, but I got stuck figuring out how to convert the GenericArray into a [u8;16] - please let me know if there's a trick I'm missing!

durin42 commented:
I'm not sure what's up with the `cargo clippy` presubmit check. Works here?

newpavlov commented on Cargo.toml:
```diff
@@ -58,18 +58,18 @@ repository = "uuid-rs/uuid"
+[dependencies.md-5]
 optional = true
-version = "0.7"
+version = "0.9.1"
```
It could be worth to start with v0.8 versions of `md-5` and `sha-1` to avoid bumping MSRV. Updating versions can be done later in a separate PR.

KodrAus replied:
@newpavlov We've actually already bumped the MSRV so there's no reason not to track the latest versions here.

newpavlov commented on src/v3.rs:
```diff
@@ -20,15 +20,14 @@ impl Uuid {
+        let computed = hasher.finalize();
 
-        let mut builder = crate::Builder::from_bytes(bytes);
+        let mut builder = crate::Builder::from_slice(&computed[..16]).unwrap();
```
IIUC you don't need the slicing here, i.e. you can replace `&computed[..16]` with `&computed`. Same goes for SHA-1.

newpavlov commented on src/v3.rs:
```diff
@@ -20,15 +20,14 @@ impl Uuid {
     /// [`NAMESPACE_X500`]: #associatedconstant.NAMESPACE_X500
     pub fn new_v3(namespace: &Uuid, name: &[u8]) -> Uuid {
-        let mut context = md5::Context::new();
+        let mut hasher = md5::Md5::new();
```
Use `use md5::{Md5, Digest};` impoer to avoid duplication of `md5` here? Same goes for SHA-1.

KodrAus commented:
It looks like it's trying to read configuration files for dependencies and is failing on one of them:

```
error reading Clippy's configuration file `C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\typenum-1.12.0\clippy.toml`
```

We can probably just remove the `clippy` check in CI if you want to remove the [`clippy.yaml`](https://github.com/uuid-rs/uuid/blob/master/.github/workflows/clippy.yml) file.

KodrAus commented:
I'm not familiar with `generic_array` but took a quick look and it looks like there's a `From` impl from `GenericArray` into ` [T; N]` if we bump our MSRV to `1.41.0`, which I'm personally ok with. `1.41.0` was released almost 12 months ago.

KodrAus commented:
Hi @durin42 :wave:

I've ended up doing this in #526. Thanks for working on this!
